lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2041,6 +2041,105 @@ namespace {
       fd->getAttrs().add(new (Impl.SwiftContext) UnsafeAttr(/*Implicit=*/true));
     }
 
+    void generateStaticFactoryForCXXForeignRef(
+        clang::ASTContext &clangCtx, ClassDecl *result,
+        clang::CXXRecordDecl *cxxRecordDecl, clang::Sema &clangSema) {
+
+      clang::IdentifierTable &idents = clangCtx.Idents;
+      clang::IdentifierInfo *generatedFuncName = &idents.get(
+          ("__returns_" + cxxRecordDecl->getNameAsString()).c_str());
+
+      // TODO/DOUBT: How to add `_Nonnull` as a qualifier on cxxRecordPtrTy
+      // return type ??
+      clang::QualType cxxRecordTy = clangCtx.getRecordType(cxxRecordDecl);
+      clang::QualType cxxRecordPtrTy = clangCtx.getPointerType(cxxRecordTy);
+      clang::FunctionProtoType::ExtProtoInfo EPI;
+      clang::QualType generatedFuncTy =
+          clangCtx.getFunctionType(cxxRecordPtrTy, {}, EPI);
+
+      // TODO: Use operatorNew directly instead of calling BuildCXXNew
```

Comment:
I think using `BuildCXXNew` is fine. True, it will repeat the call to `FindAllocationFunctions`. But I think that is better than us trying to replicate all the other logic from `BuildCXXNew` here.